lib/virtual-fs/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+use std::any::Any;
+use std::fmt;
+use std::io::{self, Read, Seek, Write};
+use std::path::Path;
+use thiserror::Error;
+use tracing::debug;
+
+pub mod host_fs;
+
+pub trait FileSystem: Send + Sync + 'static {
+    fn read_dir(&self, path: &Path) -> Result<std::fs::ReadDir, FsError>;
```

Comment:
The `std::fs::ReadDir` type is very restrictive since it's not a type that we can actually generate if our filesystem is not a host one (think on having this over a tar file, the read_dir will need to be implemented by the FileSystem implementor).

I believe we can solve this by either return a iterator of strings or a custom readdir trait.

Reply:
Yeah, it's wip